### DISCUSSION TITLE
How can I filter a collection by it's realationship?

### DISCUSSION THREAD

**Discussion by jramiroz98** [pocketbase/pocketbase#3886] [Q&A] [answered]
I want to filter the items from a collection where a field matches the id from a record in a different collection

I got my products collection and each has an id
I got the products variants collection, but I would only like to call them filtered where their product_id is the product.id
The relationship in the products variants collection is already setup, where product_id comes from the products collection, but it won't let me filter

I'm trying

	let variants = await pb.collection('variants').getFullList({ filter: `product_id="${product.id}"` });

when i just call all the variants in variants collection i can read `product_id: [ 'fhanp9008tolvkl' ],`
my result is [] even when it has a matching product_id

**Comment by ganigeorgiev** [comment, marked as answer]
Direct relation comparison works only with single relations.
From the above `product_id` output I assume that in your case it is a multiple, right?
If yes, then you can use dot-notation with "any/at-least-one" of operator. Something like:

```js
let variants = await pb.collection('variants').getFullList({
    // or alternatively if you want to use placeholders
    // filter: pb.filter("product_id.id ?= {:id}", { id: "some_id"})
    filter: `product_id.id ?= "some_id"`
});
```

**Reply by jramiroz98** [reply]
You are the best! Thank you so much!
